extension/src/providers/gotoProvider.ts

```diff
@@ -0,0 +1,145 @@
+					if (atom) {
+						selected = atom.symbol.replace(/^[']*/, '');
+					}
+				}
+			});
+		}
+		if (['(', ')', '\'', '.'].includes(selected)){
+			return;
+		}
+		try {
+			// This has a "preference" for opened and project documents for actual definitions, but will only handle variables on the opened document.
+			const locations: vscode.Location[] = [];
+			const parentContainer = rDoc.atomsForest.find(p => p instanceof Sexpression && p.contains(position)) as Sexpression;
+			const innerContainer = parentContainer?.getSexpressionFromPos(position);
+			const possibleDefun = innerContainer ? parentContainer.getParentOfSexpression(innerContainer) : null;
+			const firstChild = innerContainer?.getNthKeyAtom(0);
+			const altFirstChild = parentContainer?.getNthKeyAtom(0);
+
+			let isFunction = false;
+			if (firstChild && !(firstChild instanceof Sexpression) && selected === firstChild.symbol) {
+				isFunction = true; // most likely a function
+			} else if (altFirstChild && selected === altFirstChild.symbol) {
+				isFunction = true; // added to capture document root level function calls
+			}
+			if (isFunction && possibleDefun && /^DEFUN$|^DEFUN-Q$|^LAMBDA$/i.test(possibleDefun.getNthKeyAtom(0).symbol)) {
+				const symbolType = possibleDefun.getNthKeyAtom(0).symbol;
+				const varHeader = possibleDefun.getNthKeyAtom(symbolType.toUpperCase() === 'LAMBDA' ? 1 : 2);
+				if (innerContainer.line === varHeader.line && innerContainer.column === varHeader.column){
+					isFunction = false; // course correct for the first argument in a variable header being interpreted as a function call
+				}
+			}
+
+
+			// This condition determines the likelyhood of this being a function definition request or (else) a variable definition request
+			if (isFunction) {
+				let possible = this.findDefunMatches(selected, AutoLispExt.Documents.OpenedDocuments.concat(AutoLispExt.Documents.ProjectDocuments));
+				if (possible.length === 0) {
+					possible = this.findDefunMatches(selected, AutoLispExt.Documents.WorkspaceDocuments);
+				}
+				possible.forEach(item => {
+					locations.push(item);
+				});
+			} else if (parentContainer) {
+				this.findFirstVariableMatch(rDoc, position, selected, parentContainer).forEach(item => {
+					locations.push(item);
+				});
+			}
+			return locations;
+		} catch (error) {
+			return;	// I don't believe this requires a localized error since VSCode has a default "no definition found" response
+		}
+	}
+
+	
+	private findFirstVariableMatch(doc: ReadonlyDocument, start: vscode.Position, searchFor: string, searchIn: Sexpression): vscode.Location[] {
+		const result: vscode.Location[] = [];
+		const ucName = searchFor.toUpperCase();
+		let context = searchIn.getSexpressionFromPos(start);
+		let flag = true;
+		do {
+			const parent = searchIn.getParentOfSexpression(context);
+			const atom = parent?.getNthKeyAtom(0);
+			if (atom && /^DEFUN$|^DEFUN-Q|^LAMBDA$/i.test(atom.symbol)) {
```

Comment:
It is interesting that this is my first see this form `/^DEFUN$|^DEFUN-Q|^LAMBDA$/i `

Reply:
Is that good or bad? I see I missed a $ at the end of DEFUN-Q. What do you normally do to say something has to _completely_ match? I know they would need to be in parenthesis if there are "other" pieces I was trying to also match.

Reply:
It is good! I think maybe give more readable name for this regex `/^DEFUN$|^DEFUN-Q|^LAMBDA$/i`